src/java_bytecode/java_object_factory.cpp

```diff
@@ -623,32 +785,32 @@ void java_object_factoryt::gen_nondet_pointer_init(
   // vector of assignments that create a new object (recursively initializes it)
   // and asign to `expr` the address of such object
   code_blockt non_null_inst;
-  gen_pointer_target_init(
-    non_null_inst,
+
+  if(add_nondet_string_pointer_initialization(
```

Comment:
This structure is a bit odd since it conflates "not a string, this function isn't applicable" with an actual error. Suggest splitting into is-string and init-string functions.